common/src/test/java/org/apache/drill/test/BaseTest.java

```diff
@@ -15,24 +15,25 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.drill.exec.expr;
+package org.apache.drill.test;
 
-import org.apache.drill.BaseTestQuery;
-import org.apache.drill.common.util.TestTools;
-import org.junit.Test;
+import org.apache.drill.common.util.GuavaPatcher;
+import org.apache.drill.common.util.ProtobufPatcher;
 
-public class TestPrune extends BaseTestQuery {
+public class BaseTest {
```

Comment:
The term `BaseTest` is too generic: soon this will contain all manner of common code. Better solutions are:

* MapRTestBase and HBaseTestBase to patch the code as needed for just those clients.
* Patchers as a service added to tests where needed, such as is done with the "dirTestWatcher"
* The patch-on-load approach suggested below.

The approach here is really far too much of a hack to maintain.

Also, this patching means that code running in tests operates in a different environment than the same code running in production. That will become the cause of very mysterious bugs.

Can we rethink our approach?

Reply:
The original problem was caused by tests that didn't require patching, they just loaded a Guava class needed to be patched for someone else, an there is no way we can patch a class that was already loaded. Considering this we should apply patching as early as possible.

Reply:
Patching is done for production code also, so with patching, we make testing env closer to the production one. Also, pathing is required for a newer hadoop libraries version which is used in exec module, so comment in `BaseTest` perhaps should be updated a little bit.